### DISCUSSION TITLE
Double camera entry

### DISCUSSION THREAD

**Discussion by evripidis** [aaronwmorris/indi-allsky#1155] [General]
@aaronwmorris Is this something that came up after the zwo drivers and the latest indi issue that we were talking about days ago?

<img width="295" alt="Στιγμιότυπο οθόνης 2024-02-27, 12 47 06" src="https://github.com/aaronwmorris/indi-allsky/assets/216827/2d77cfa8-5f30-4a44-98a5-490d2eb3c041">

**Comment by aaronwmorris** [comment]
The name of the camera changed with the upgrade of INDI.  indi-allsky uses the name as the primary identifier of a camera, so if you insert a different camera, the images will be associated with the different camera.

I can give you some SQL if you want to change the name of camera in the DB so that it sends new images to the old entry.

**Reply by evripidis** [reply]
The only thing that I've done is to update the system. My 224 is connected to the system for months (I'm not even there, I remote login there) and I certainly do not have 2 cameras attached.

As far as I can understand all my data are under the CCD version for the driver. I do not mind if it says CCD, the only thing that bothers me is not to lose my data etc like the changes we had in Kstars/Ekos (we've discussed this subject there too).

As far as we know the CCD version is the older one, so I would love to know if it will be removed one day from the project.

**Reply by aaronwmorris** [reply]
Use these steps to update the old camera entry to the new name.

1. Stop indi-allsky
    - `systemctl --user stop indi-allsky`
2. Rename new camera
    - `sqlite3 /var/lib/indi-allsky/indi-allsky.sqlite "UPDATE camera SET name='ZWO CCD ASI224MC rename' WHERE name='ZWO CCD ASI224MC';"`
3. Hide new camera
    - `sqlite3 /var/lib/indi-allsky/indi-allsky.sqlite "UPDATE camera SET hidden=1 WHERE name='ZWO CCD ASI224MC rename';"`
4. Rename old camera
    - `sqlite3 /var/lib/indi-allsky/indi-allsky.sqlite "UPDATE camera SET name='ZWO CCD ASI224MC' WHERE name='ZWO ASI224MC';"`
5.  Start indi-allsky
    - `systemctl --user start indi-allsky`

**Reply by evripidis** [reply]
Thank Aaron "it did worked" but FYI it still says 2 cameras available despite that there is only one option.

Already rebooted the unit and tried different browser.

<img width="280" alt="Στιγμιότυπο οθόνης 2024-02-27, 18 55 17" src="https://github.com/aaronwmorris/indi-allsky/assets/216827/db9ca44a-4a2c-4c89-8d22-3fa19aa40e9c">


*** EDIT. Not working.

<img width="1023" alt="Στιγμιότυπο οθόνης 2024-02-27, 19 11 25" src="https://github.com/aaronwmorris/indi-allsky/assets/216827/90eb202f-7887-486c-845c-3fc575814d04">

**Reply by aaronwmorris** [reply]
Fixed in #1159 

I have no idea how the query was even working in the first place.

```diff
         context['camera_count'] = IndiAllSkyDbCameraTable.query\
-            .order_by(IndiAllSkyDbCameraTable.hidden == sa_false())\
+            .filter(IndiAllSkyDbCameraTable.hidden == sa_false())\
             .count()
```

**Reply by aaronwmorris** [reply]
Clear your cookies, it may be stuck on the hidden camera in the session.

**Reply by evripidis** [reply]
I had to update the system. Up and running again.

Thank you @aaronwmorris !
